crmsh/bootstrap.py

```diff
@@ -509,6 +519,45 @@ def _get_sbd_device_from_config(self):
         else:
             return None
 
+    def _restart_cluster_on_needed(self):
+        """
+        Try to configure sbd resource, restart cluster on needed
+        """
+        if not utils.has_resource_running():
+            status("Restarting cluster service")
+            utils.cluster_run_cmd("crm cluster restart")
+            wait_for_cluster()
+            self.configure_sbd_resource()
+        else:
+            warn("To start sbd.service, need to restart cluster service manually on each node")
+            if self.diskless_sbd:
+                warn("Then run \"crm configure property stonith-enabled=true stonith-watchdog-timeout=5s\" on any node")
+            else:
+                self.configure_sbd_resource()
+
+    def _enable_sbd_service(self):
+        """
+        Try to enable sbd service
+        """
+        if _context.cluster_is_running:
+            # in sbd stage, enable sbd.service on cluster wide
+            utils.cluster_run_cmd("systemctl enable sbd.service")
+            self._restart_cluster_on_needed()
+        else:
+            # in init process
+            invoke("systemctl enable sbd.service")
+
+    def _warn_diskless_sbd(self, peer=None):
+        """
+        Give warning when configuring diskless sbd
+        """
+        if self.diskless_sbd and _context.cluster_is_running or peer:
```

Comment:
No idea how the priorities of the operators `and` and `or` are handled in python like in here :-) At least the readability is not very clear.

Reply:
Changed as `if (self.diskless_sbd and _context.cluster_is_running) or peer`